lib/rmt/mirror.rb

```diff
@@ -148,14 +145,15 @@ def mirror_data(primary_files, deltainfo_files)
     @downloader.destination_dir = @repository_dir
     @downloader.cache_dir = nil
 
+    failed_downloads = []
     deltainfo_files.each do |filename|
       parser = RMT::Rpm::DeltainfoXmlParser.new(
         File.join(@temp_metadata_dir, filename),
         @mirror_src
       )
       parser.parse
       to_download = parsed_files_after_dedup(@repository_dir, parser.referenced_files)
-      @downloader.download_multi(to_download) unless to_download.empty?
+      failed_downloads += @downloader.download_multi(to_download, ignore_errors: true) unless to_download.empty?
```

Comment:
I would prefere `<<` operator, since it does not create new object

Reply:
It does a different thing. `+` merges two arrays into a one-dimensional array, `<<` will create a two-dimensional array.

Reply:
you are right. but what if we use `Array#concat` ?

Reply:
Yep, good suggestion, `concat` would work.